.github/workflows/regression.yml

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 GlobalFoundries PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: regression testing
+
+# Prevent keeping resources busy when a branch/PR is updated
+# https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: ${{ github.repository }}-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+env:
+  GITHUB_TOKEN: ${{ github.token }}
+  DOCKER_BUILDKIT: 1
+  IMAGE: ghcr.io/mabrains/pdk_regression
+
+jobs:
+  regression:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { tool: klayout, pdk: drc, test: main }
+          - { tool: klayout, pdk: drc, test: switch }
+          - { tool: klayout, pdk: lvs, test: main }
+          - { tool: klayout, pdk: lvs, test: switch }
+
+    name: ${{ matrix.pdk }} | ${{ matrix.test }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - run: docker pull "$IMAGE"
+      - name: Testing ${{ matrix.type }} for ${{ matrix.test }}
+        run: >
+          docker run --rm
+          -v $(pwd):/wrk -w /wrk/${{ matrix.tool }}/${{ matrix.pdk }}/testing
+          -e PDK_ROOT="/wrk/${{ matrix.tool }}"
+          -e PDK='${{ matrix.pdk }}'
+          "$IMAGE"
+          make test-"$(python -c 'print("${{ matrix.pdk }}".upper())')"-${{ matrix.test }}
```

Comment:
should we add a dummy Makefile, otherwise this will fail the CI.

Reply:
@proppy I remembered, that we agreed the CI will always start with errors at the first few PRs and will be fixed and start to be clean as we go. I thought that was acceptable according to our agreement in the meeting. Do you want me to add a base make file in this PR as well?

Reply:
I think that's less critical to land the CI first now that we are starting from an empty repo, so I'd be in favor of include "some checks" to see that the CI is working.

@jeffdi might also have some input there.

Reply:
@proppy I have changed the code and added the `makefile` in the main part as shown in the last commit. I have added empty targets for now. I'll change that later.